source/renderer/app/components/wallet/WalletImporter.js

```diff
@@ -0,0 +1,255 @@
+    defaultMessage: '!!!Drop your "secret.key" file here or click to choose',
+    description: 'Hint for the secrets key file upload field on the wallet importer page.'
+  },
+  passwordsListLabel: {
+    id: 'wallet.importer.passwordsListLabel',
+    defaultMessage: '!!!Passwords list',
+    description: 'Label "Passwords list" on the wallet importer page.'
+  },
+  passwordsListHint: {
+    id: 'wallet.importer.passwordsListHint',
+    defaultMessage: '!!!Enter your potential wallet passwords line by line',
+    description: 'Hint for the passwords list field on the wallet importer page.'
+  },
+  spendingPasswordLabel: {
+    id: 'wallet.restore.dialog.spendingPasswordLabel',
+    defaultMessage: '!!!Enter password',
+    description: 'Label for the "Wallet password" input in the wallet restore dialog.',
+  },
+  passwordFieldPlaceholder: {
+    id: 'wallet.restore.dialog.passwordFieldPlaceholder',
+    defaultMessage: '!!!Password',
+    description: 'Placeholder for the "Password" inputs in the wallet restore dialog.',
+  },
+});
+
+type Props = {};
+
+@observer
+export default class WalletImporter extends Component<Props> {
+
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  form = new ReactToolboxMobxForm({
+    fields: {
+      keyFile: {
+        label: this.context.intl.formatMessage(messages.keyFileLabel),
+        placeholder: this.context.intl.formatMessage(messages.keyFileHint),
+        type: 'file',
+      },
+      passwords: {
+        label: this.context.intl.formatMessage(messages.passwordsListLabel),
+        placeholder: this.context.intl.formatMessage(messages.passwordsListHint),
+        value: '',
+      },
+      walletHashes: {
+        value: [],
+      },
+      confirmedPasswordHashes: {
+        value: [],
+      },
+      spendingPassword: {
+        type: 'password',
+        label: this.context.intl.formatMessage(messages.spendingPasswordLabel),
+        placeholder: this.context.intl.formatMessage(messages.passwordFieldPlaceholder),
+        value: '',
+      },
+    },
+  }, {
+    options: {
+      validateOnChange: true,
+      validationDebounceWait: FORM_VALIDATION_DEBOUNCE_WAIT,
+    },
+  });
+
+  submit() {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const spendingPasswordField = form.$('spendingPassword');
+    console.log(spendingPasswordField.value);
+    this.testPassword(spendingPasswordField.value);
+  }
+
+  checkPassword(passphraseHash, passhash) {
+    var bits = passphraseHash.split("|");
+    var logN = parseInt(bits[0]);
+    var r = parseInt(bits[1]);
+    var p = parseInt(bits[2]);
+    var salt = Buffer.from(bits[3],"base64");
+    var realhash = Buffer.from(bits[4],"base64");
+
+    // from nodejs native crypto (which is missing in renderer proc)
+    // var hash2 = scryptSync(cbor.encode(passhash), salt, 32, { N: 2 ** logN, r: r, p: p });
+    // try the scryptsy from npm (works, but worse performance)
```

Comment:
`scryptsy` has worse performance, it would be better to use the native `crypto.scryptSync`
but that isnt available in the renderer process